test/util/testnode/utils.go

```diff
@@ -118,6 +123,132 @@ func mustGetFreePort() int {
 	return port
 }
 
+// IsPortAvailable checks if a port is available on localhost.
+func IsPortAvailable(port int) bool {
+	addr := fmt.Sprintf("127.0.0.1:%d", port)
+	conn, err := net.DialTimeout("tcp", addr, 100*time.Millisecond)
+	if err != nil {
+		// Port is available if we can't connect to it
+		return true
+	}
+	conn.Close()
+	return false
+}
+
+// KillProcessOnPort attempts to kill processes listening on the specified port.
+// It returns an error if the port cleanup fails.
+func KillProcessOnPort(port int) error {
```

Comment:
delete and don't use this, as I think this is too harsh. instead we should just retry finding a truly open port until we are 100% confident that we have one